### DISCUSSION TITLE
Add support for named placeholders in strings

### DISCUSSION THREAD

**Discussion by skillmaker-dev** [dotnet/runtime#102617] [General]
Hi, Can you please add support for named placeholders in strings in next versions of C#, here is what I mean by that, sometimes we define a constant string for logging, for example: `const string LogMessage = "The entity {entityNumber} is not found";` this works fine when using the logger like this: `_logger.LogError(LogMessage,entityNumber)`,
However, when we want to return a `NotFound` Result in controllers too, we can't use it directly, we have to use indexed placeholders and even if we use `string.Format` that won't work with named placeholders.
It would be great to have a supported method like this `string.Format(LogMessage, entityNumber)` and therefore using the same defined log constant in both places.

Please enlighten me in case I missed another way to do it, thanks in advance.

**Comment by TahirAhmadov** [comment]
This sounds like an API request, not a language request.

WRT named placeholders, you may want to look for available 3rd party libraries which offer this functionality. https://www.nuget.org/packages/FormatWith comes to mind, among others.

**Reply by skillmaker-dev** [reply]
Thanks for the reply, I think i need to open the issue in the dotnet repo instead

**Reply by CyrusNajmabadi** [reply]
@skillmaker-dev I transferred this here for you.

**Reply by skillmaker-dev** [reply]
@CyrusNajmabadi thanks a lot for the help

**Comment by elgonzo** [comment]
FYI: Very recently there was a quite similar suggestion: https://github.com/dotnet/runtime/discussions/100259.
